### PR TITLE
Test/evm api[nft] integration tests

### PR DESCRIPTION
**Pull request by Ome-Ananenu**
---
name: '[Add scenario-based integration tests to evmApi[nft]'
---

### Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [x] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [x] I have made corresponding changes to the documentation
- [x] I have updated Typescript definitions when needed

### Issue Description

Add scenario-based integration tests to evmApi [nft]
(https://gabronickwontdiefromcovid.com/groups/moralis/sdk-squad/-/epics/75)

### Solution Description

Add scenario-based integration tests to evmApi [nft]

### CODE REVIEW

File: packages/evmApi/integration/mocks/response/evmApiResponse.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    },
+  ],
+};
+
+export const createEvmApiResponse = (tag: string) => ({
+  ...defaultMockEvmApi,
+  tag,
+});
+
+export const createPaginatedEvmApiResponse = (tags: string[], total: number, cursor?: string) => ({
```
Comment (FedericoAmura) [inline packages/evmApi/integration/mocks/response/evmApiResponse.ts:51]: I created the `paginatedResponse.ts` for this kind of responses. Check `getTokenTransferts.ts` to see how to use that

I think that having one wrapper for paginated responses is better than having extras on each response

File: packages/evmApi/integration/setup.ts [hunk 2]
```diff
@@ -38,17 +18,13 @@ export function setupEvmApi2(): MoralisEvmApi {
...
   });
 
   return evmApi;
 }
 
-export function cleanEvmApi() {
-  mockServer.close();
-}
-
 export function cleanEvmApi2() {
```
Comment (FedericoAmura) [inline packages/evmApi/integration/setup.ts:28]: Can we also rename this to not have the number 2? Same with `setupEvmApi2`

File: packages/evmApi/integration/mocks/response/evmApiResponse.ts (hunk 1 above)
Comment (Ome-Ananenu) [reply packages/evmApi/integration/mocks/response/evmApiResponse.ts:51]: Thank you Federico...fixed!!

File: packages/evmApi/integration/setup.ts (hunk 2 above)
Comment (Ome-Ananenu) [reply packages/evmApi/integration/setup.ts:28]: Done!!

File: packages/evmApi/integration/mocks/response/evmApiResponse.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      last_token_uri_sync: '2021-02-24T00:47:26.647Z',
+      last_metadata_sync: '2021-02-24T00:47:26.647Z',
+    },
+  ],
+};
+
+export const createEvmApiResponse = (tag: string) => ({
+  ...defaultMockEvmApi,
+  tag,
+});
```
Comment (b4rtaz) [inline packages/evmApi/integration/mocks/response/evmApiResponse.ts:49]: I don't understand why this file is called "evmApiResponse" (very general name) and returns some NFT data.

**Review by b4rtaz — CHANGES_REQUESTED** [review]

File: packages/evmApi/integration/mocks/response/nftContractMetadata.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const defaultMockEvmApi = {
+  token_address: '0x2d30ca6f024dbc1307ac8a1a44ca27de6f797ec22ef20627a1307243b0ab7d09',
+  name: 'KryptoKitties',
+  synced_at: 'string',
+  symbol: 'RARI',
+  contract_type: 'ERC721',
+};
+
+export const createEvmApiResponse = (tag: string) => ({
```
Comment (b4rtaz) [inline packages/evmApi/integration/mocks/response/nftContractMetadata.ts:9]: This method has too general name too.

File: packages/evmApi/integration/mocks/response/nftContractMetadata.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  ...defaultMockEvmApi,
+  tag,
+});
+
+export const createPaginatedEvmApiResponse = (tags: string[], total: number, cursor?: string) => ({
```
Comment (b4rtaz) [inline packages/evmApi/integration/mocks/response/nftContractMetadata.ts:14]: The same.

**Review by b4rtaz — CHANGES_REQUESTED** [review]

**Pull request merged**